logical/framework/openapi_test.go

```diff
@@ -0,0 +1,424 @@
+	"github.com/hashicorp/vault/helper/jsonutil"
+	"github.com/hashicorp/vault/helper/openapi"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/version"
+)
+
+/*
+
+var reqdRe = regexp.MustCompile(`\(?\?P<(\w+)>[^)]*\)?`) // Capture required parameters, e.g. "(?P<name>regex)"
+var optRe = regexp.MustCompile(`(?U)\(.*\)\?`)           // Capture optional path elements in ungreedy (?U) fashion, e.g. "(leases/)?renew"
+var altRe = regexp.MustCompile(`\((.*)\|(.*)\)`)         // Capture alternation elements, e.g. "(raw/?$|raw/(?P<path>.+))"
+var pathFieldsRe = regexp.MustCompile(`{(\w+)}`)         // Capture OpenAPI-style named parameters, e.g. "lookup/{urltoken}",
+var cleanCharsRe = regexp.MustCompile("[()^$?]")         // Set of regex characters that will be stripped during cleaning
+var cleanSuffixRe = regexp.MustCompile(`/\?\$?$`)        // Path suffix patterns that will be stripped during cleaning
+var wsRe = regexp.MustCompile(`\s+`)                     // Match whitespace, to be compressed during cleaning
+*/
+
+func TestOpenAPI_Regex(t *testing.T) {
+	t.Run("Required", func(t *testing.T) {
+		tests := []struct {
+			input    string
+			captures []string
+		}{
+			{`/foo/bar/(?P<val>.*)`, []string{"val"}},
+			{`/foo/bar/` + GenericNameRegex("val"), []string{"val"}},
+			{`/foo/bar/` + GenericNameRegex("first") + "/b/" + GenericNameRegex("second"), []string{"first", "second"}},
+			{`/foo/bar`, []string{}},
+		}
+
+		for _, test := range tests {
+			result := reqdRe.FindAllStringSubmatch(test.input, -1)
+			if len(result) != len(test.captures) {
+				t.Fatalf("Capture error (%s): expected %d matches, actual: %d", test.input, len(test.captures), len(result))
+			}
+
+			for i := 0; i < len(result); i++ {
+				if result[i][1] != test.captures[i] {
+					t.Fatalf("Capture error (%s): expected %s, actual: %s", test.input, test.captures[i], result[i][1])
+				}
+			}
+		}
+	})
+	t.Run("Optional", func(t *testing.T) {
+		input := "foo/(maybe/)?bar"
+		expStart := len("foo/")
+		expEnd := len(input) - len("bar")
+
+		match := optRe.FindStringIndex(input)
+		if diff := deep.Equal(match, []int{expStart, expEnd}); diff != nil {
+			t.Fatal(diff)
+		}
+
+		input = "/foo/maybe/bar"
+		match = optRe.FindStringIndex(input)
+		if match != nil {
+			t.Fatalf("Expected nil match (%s), got %+v", input, match)
+		}
+	})
+	t.Run("Alternation", func(t *testing.T) {
+		input := `(raw/?$|raw/(?P<path>.+))`
+
+		matches := altRe.FindAllStringSubmatch(input, -1)
+		exp1 := "raw/?$"
+		exp2 := "raw/(?P<path>.+)"
+		if matches[0][1] != exp1 || matches[0][2] != exp2 {
+			t.Fatalf("Capture error. Expected %s and %s, got %v", exp1, exp2, matches[0][1:])
+		}
+
+		input = `/foo/bar/` + GenericNameRegex("val")
+
+		matches = altRe.FindAllStringSubmatch(input, -1)
+		if matches != nil {
+			t.Fatalf("Expected nil match (%s), got %+v", input, matches)
+		}
+	})
+	t.Run("Path fields", func(t *testing.T) {
+		input := `/foo/bar/{inner}/baz/{outer}`
```

Comment:
I don't think our path definitions use this format, when would this regex be used?

Reply:
If you mean multiple path parameters, there are not many places but still a few, e.g.: https://github.com/hashicorp/vault/blob/master/builtin/logical/transit/path_export.go#L29